core/src/com/unciv/ui/images/ImageGetter.kt

```diff
@@ -252,11 +252,24 @@ object ImageGetter {
         return iconGroup
     }
 
-    fun getConstructionImage(construction: String): Image {
-        if (ruleset.buildings.containsKey(construction)) return getImage("BuildingIcons/$construction")
-        if (ruleset.units.containsKey(construction)) return getUnitIcon(construction)
-        if (construction == "Nothing") return getImage("OtherIcons/Sleep")
-        return getStatIcon(construction)
+    fun getPortraitImage(construction: String, size: Float): Group {
+        if (ruleset.buildings.containsKey(construction)) {
+            val buildingPortraitLocation = "BuildingPortraits/$construction"
+            return if (imageExists(buildingPortraitLocation)) {
+                getImage(buildingPortraitLocation).toGroup(size)
+            } else
+                getImage("BuildingIcons/$construction").surroundWithCircle(size)
```

Comment:
```suggestion
                getImage(buildingPortraitLocation).surroundWithCircle(size)
```

Reply:
This one is not possible:
`buildingPortraitsLocation` leads to `BuildingPortraits/`
and in else we lead to `BuildingIcons/`

Reply:
Right you are